tools/release_builder.py

```diff
@@ -74,33 +74,81 @@ def _run_cmd(self, cmd, exit_on_fail=True, dry_run_return = 0):
             return ret
 
     def _download_unpack_stub_universe(self, scratchdir):
-        '''Returns the path to the directory in the stub universe.'''
-        local_zip_path = os.path.join(scratchdir, self._stub_universe_url.split('/')[-1])
-        result = urllib.request.urlopen(self._stub_universe_url)
-        dlfile = open(local_zip_path, 'wb')
-        dlfile.write(result.read())
-        dlfile.flush()
-        dlfile.close()
-        zipin = zipfile.ZipFile(local_zip_path, 'r')
-        badfile = zipin.testzip()
-        if badfile:
-            raise Exception('Bad file {} in downloaded {} => {}'.format(
-                badfile, self._stub_universe_url, local_zip_path))
-        zipin.extractall(scratchdir)
-        # check for stub-universe-pkgname/repo/packages/P/pkgname/0/:
-        pkgdir_path = os.path.join(
-            scratchdir,
-            'stub-universe-{}'.format(self._pkg_name),
-            'repo',
-            'packages',
-            self._pkg_name[0].upper(),
-            self._pkg_name,
-            '0')
-        if not os.path.isdir(pkgdir_path):
-            raise Exception('Didn\'t find expected path {} after unzipping {}'.format(
-                pkgdir_path, local_zip_path))
-        os.unlink(local_zip_path)
-        return pkgdir_path
+        '''Returns the path to the package directory in the stub universe.'''
+        stub_universe_file = urllib.request.urlopen(self._stub_universe_url)
+
+        stub_universe_extension = os.path.splitext(self._stub_universe_url)[1]
+        if stub_universe_extension == '.zip':
```

Comment:
Nitpick: I'd rather see these cases split into functions. Don't have a strong argument for it other than stylistic.